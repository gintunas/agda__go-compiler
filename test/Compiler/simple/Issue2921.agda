
open import Agda.Builtin.IO
open import Agda.Builtin.Unit
open import Agda.Builtin.Nat

data Unit : Set where
  c : ⊤ → Nat → ⊤ → Unit

-- Note that the ⊤ arguments do not get erased and should be present in
-- the Haskell version of Unit.
{-# FOREIGN GHC data Unit = Unit () Integer () #-}
{-# COMPILE GHC Unit = data Unit (Unit) #-}

postulate print : Nat → IO ⊤
{-# COMPILE GHC print = print #-}
{-# COMPILE JS  print = function (x) { return function(cb) { process.stdout.write(x + "\n"); cb(0) } } #-}

foo : Unit → IO ⊤
foo (c _ n _) = print n

main : IO ⊤
main = foo (c _ 12345 _)
