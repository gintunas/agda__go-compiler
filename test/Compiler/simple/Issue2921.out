EXECUTED_PROGRAM

ret > ExitSuccess
out > 12345
out >
